tests/dialects/test_arith.py

```diff
@@ -128,6 +129,29 @@ def test_Cmpi_from_mnemonic(self, input: str):
         _ = CmpiOp(self.a, self.b, input)
 
 
+def test_constant_construction():
+    c1 = ConstantOp(IntegerAttr(1, i32))
+    assert c1.value.type == i32
+
+    c2 = ConstantOp(IntegerAttr(1, i32), i64)
+    assert c2.value.type == i64
```

Comment:
Is there a use case where this behaviour is desirable

Reply:
What behaviour exactly?

Reply:
Passing in an typed attribute and a different type, and having the constructor make a new attribute with the new type

Reply:
There are some obscure uses (one in CSL).

Reply:
I'm not sure it makes sense to support this

Reply:
patched the ones that were left (in experimental)